pkg/provisioner/ironic/testserver/server.go

```diff
@@ -0,0 +1,104 @@
+package testserver
```

Comment:
I think it's really a good idea to have a dedicated file for a test helper, as it supports reuse in different tests. What about defining a dedicated folder to store test helpers, in case in future some others will be defined? Not sure if there's a consolidated convention for that.

Reply:
Do you mean to move this to something like `/pkg/fixtures`?

Reply:
Yes (and keeping also the convention to have the suffix `test` in the package name - ie like `httptest` - to clearly state that it's not production code)

Reply:
I could see renaming the package, but it's only going to be used by the ironic provisioner tests so I'm not sure I see the benefit of moving it further away from that test suite.

Reply:
Don't want to anticipate too much the changes, so until they're going to be used just here then I think it's fine to delay moving them into another general folder